### PR TITLE
Smz3 gt fill bias fix

### PR DESCRIPTION
**Pull request by lordlou**
- fixed bias toward "nice" pool of SMZ3 when prefilling GT with junk

### CODE REVIEW

File: worlds/smz3/__init__.py [hunk 1]
```diff
@@ -479,20 +479,27 @@ def write_spoiler(self, spoiler_handle: TextIO):
             self.world.spoiler.unreachables.update(self.unreachable)
 
     def JunkFillGT(self):
+        poolLength = len(self.world.itempool)
+        junkPool = [(i, self.world.itempool[i]) for i in range(0, poolLength) 
+                    if self.world.itempool[i].classification in (ItemClassification.filler, ItemClassification.trap)]
```
Comment (black-sliver) [inline worlds/smz3/__init__.py:484]: would be easier on the eyes to only collect the indices of candidates, and use `world.itempool[junk[i]]` below

File: worlds/smz3/__init__.py [hunk 2]
```diff
@@ -479,20 +479,27 @@ def write_spoiler(self, spoiler_handle: TextIO):
...
                 # start looking at a random starting index and loop at start if no match found
                 start = self.world.random.randint(0, poolLength)
                 for off in range(0, poolLength):
                     i = (start + off) % poolLength
-                    if self.world.itempool[i].classification in (ItemClassification.filler, ItemClassification.trap) \
-                            and loc.can_fill(self.world.state, self.world.itempool[i], False):
-                        itemFromPool = self.world.itempool.pop(i)
+                    if i not in toRemove and loc.can_fill(self.world.state, junkPool[i][1], False):
+                        itemFromPool = junkPool[i][1]
+                        toRemove.append(junkPool[i][0])
```
Comment (black-sliver) [inline worlds/smz3/__init__.py:495]: you check for `i` in toRemove, but append `junkPool[i][0] `

**Review by black-sliver — CHANGES_REQUESTED** [review]
At this point I'm wondering if shuffling junk pool and popping from it wouldn't be better.
@Berserker66  is probably against that, but `random.shuffle(list(range(0,100000)))` only takes 30ms on my machine, so maybe that's fine?

(that's basically what i do in evermore to replace junk with traps. i do this *before* submitting it to the pool however, so it's a fixed amount of items)

File: worlds/smz3/__init__.py (hunk 1 above)
Comment (lordlou) [reply worlds/smz3/__init__.py:484]: Done in https://github.com/ArchipelagoMW/Archipelago/pull/886

File: worlds/smz3/__init__.py (hunk 2 above)
Comment (lordlou) [reply worlds/smz3/__init__.py:495]: Fixed in https://github.com/ArchipelagoMW/Archipelago/pull/886

**Comment by lordlou** [comment]
I'm closing this PR as I've included it (and done proposed change/fix) in https://github.com/ArchipelagoMW/Archipelago/pull/886

**Pull request closed without merging**